NPE when using Service resource fragment

### Description
<!--
Thank you for trying out Eclipse JKube!

Please fill out the following information to help us in analyzing the issue, but feel free to skip it if you don't have the information at hand or if it does not apply. Please remove every section which does not apply to the issue.
-->

### Info

* Eclipse JKube version : 1.3.0
* Maven version (`mvn -v`) :
```
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: /usr/local/Cellar/maven/3.8.1/libexec
Java version: 11.0.2, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/openjdk-11.0.2.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.16", arch: "x86_64", family: "mac"
```
* Kubernetes / Red Hat OpenShift setup and version :
```
Client Version: openshift-clients-4.6.0-202006250705.p0-166-g299b6af53
Server Version: 4.7.0
Kubernetes Version: v1.20.0+bd9e442
```
* If it's a bug, how to reproduce :
just create a Service fragment in the `src/main/jkube/service.yml`
with this content:
```yaml
spec:
  ports:
    - name: http
```
* If it's a feature request, what is your use case :

* Sample Reproducer Project : *[GitHub Clone URL]*

Looking at the line where the NPE is thrown: https://github.com/eclipse/jkube/blob/v1.3.0/jkube-kit/enricher/specific/src/main/java/org/eclipse/jkube/kit/enricher/specific/ServiceDiscoveryEnricher.java#L100

The plugin expects the `port` as a mandatory property in the fragment. In my case I just wanna make sure the generated Service port has the right `name` (eg. `http` which is expected by the Istio Service Mesh). The `port`number is supposed to be filled automatically by the plugin.

NPE log:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:00 min
[INFO] Finished at: 2021-05-21T15:46:21-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.3.0:resource (default-cli) on project business-application-service: Execution default-cli of goal org.eclipse.jkube:openshift-maven-plugin:1.3.0:resource failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.3.0:resource (default-cli) on project business-application-service: Execution default-cli of goal org.eclipse.jkube:openshift-maven-plugin:1.3.0:resource failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.eclipse.jkube:openshift-maven-plugin:1.3.0:resource failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.NullPointerException
    at org.eclipse.jkube.kit.enricher.specific.ServiceDiscoveryEnricher.addAnnotations (ServiceDiscoveryEnricher.java:100)
    at org.eclipse.jkube.kit.enricher.specific.ServiceDiscoveryEnricher$1.visit (ServiceDiscoveryEnricher.java:89)
    at org.eclipse.jkube.kit.enricher.specific.ServiceDiscoveryEnricher$1.visit (ServiceDiscoveryEnricher.java:85)
    at io.fabric8.kubernetes.api.builder.BaseFluent.acceptInternal (BaseFluent.java:135)
    at io.fabric8.kubernetes.api.builder.BaseFluent.accept (BaseFluent.java:110)
    at io.fabric8.kubernetes.api.builder.BaseFluent.accept (BaseFluent.java:13)
    at io.fabric8.kubernetes.api.builder.BaseFluent.acceptInternal (BaseFluent.java:132)
    at io.fabric8.kubernetes.api.builder.BaseFluent.accept (BaseFluent.java:110)
    at org.eclipse.jkube.kit.enricher.specific.ServiceDiscoveryEnricher.create (ServiceDiscoveryEnricher.java:85)
    at org.eclipse.jkube.maven.plugin.enricher.DefaultEnricherManager.lambda$createDefaultResources$1 (DefaultEnricherManager.java:72)
    at org.eclipse.jkube.maven.plugin.enricher.DefaultEnricherManager.loop (DefaultEnricherManager.java:105)
    at org.eclipse.jkube.maven.plugin.enricher.DefaultEnricherManager.createDefaultResources (DefaultEnricherManager.java:71)
    at org.eclipse.jkube.maven.plugin.enricher.DefaultEnricherManager.createDefaultResources (DefaultEnricherManager.java:65)
    at org.eclipse.jkube.kit.resource.service.DefaultResourceService.generateAppResources (DefaultResourceService.java:92)
    at org.eclipse.jkube.kit.resource.service.DefaultResourceService.generateResources (DefaultResourceService.java:59)
    at org.eclipse.jkube.maven.plugin.mojo.build.ResourceMojo.generateResources (ResourceMojo.java:345)
    at org.eclipse.jkube.maven.plugin.mojo.build.ResourceMojo.executeInternal (ResourceMojo.java:232)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractJKubeMojo.execute (AbstractJKubeMojo.java:97)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,6 +21,7 @@ Usage:
 ./scripts/extract-changelog-for-version.sh 1.3.37 5
 ```
 ### 1.4.0-SNAPSHOT
+* Fix #704: NPE when using Service fragment with no port specified
 * Fix #705: JIB assembly works on Windows
 
 ### 1.3.0

--- a/jkube-kit/enricher/specific/src/main/java/org/eclipse/jkube/kit/enricher/specific/ServiceDiscoveryEnricher.java
+++ b/jkube-kit/enricher/specific/src/main/java/org/eclipse/jkube/kit/enricher/specific/ServiceDiscoveryEnricher.java
@@ -35,6 +35,7 @@ import java.io.File;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 
 public class ServiceDiscoveryEnricher extends BaseEnricher {
     private static final String ENRICHER_NAME = "jkube-service-discovery";
@@ -97,6 +98,8 @@ public class ServiceDiscoveryEnricher extends BaseEnricher {
             List<ServicePort> ports = serviceBuilder.buildSpec().getPorts();
             if (! ports.isEmpty()) {
                 ServicePort firstServicePort = ports.iterator().next();
+                Objects.requireNonNull(firstServicePort.getPort(),
+                        String.format("Service %s .spec.ports[0].port: required value", serviceBuilder.buildMetadata().getName()));
                 port = firstServicePort.getPort().toString();
                 log.info("Using first mentioned service port '%s' " , port);
             } else {

--- a/jkube-kit/enricher/specific/src/test/java/org/eclipse/jkube/kit/enricher/specific/ServiceDiscoveryEnricherTest.java
+++ b/jkube-kit/enricher/specific/src/test/java/org/eclipse/jkube/kit/enricher/specific/ServiceDiscoveryEnricherTest.java
@@ -26,7 +26,10 @@ import mockit.Mocked;
 import org.junit.Before;
 import org.junit.Test;
 
+import javax.validation.constraints.Null;
+
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
 
 public class ServiceDiscoveryEnricherTest {
 
@@ -113,6 +116,24 @@ public class ServiceDiscoveryEnricherTest {
     public void testSchemeAnnotation() {
         enricher.addAnnotations(builder);
         assertAnnotation(scheme, "scheme");
+    }
+
+    @Test
+    public void testServiceWithNullPort() {
+        // Given
+        ServiceBuilder serviceBuilder = new ServiceBuilder()
+                .withNewMetadata().withName("test-svc").endMetadata()
+                .withNewSpec()
+                .addNewPort()
+                .withName("foo")
+                .endPort()
+                .endSpec();
+
+        // When
+        NullPointerException npe = assertThrows(NullPointerException.class, () -> enricher.addAnnotations(serviceBuilder));
+
+        // Then
+        assertEquals("Service test-svc .spec.ports[0].port: required value", npe.getMessage());
     }
 
     private void assertAnnotation(String expectedValue, String annotation) {

